### DISCUSSION TITLE
Why have both `socket.connected` and `socket.disconnected`?

### DISCUSSION THREAD

**Discussion by ryanwc** [socketio/socket.io#4168] [Q&A] [answered]
The Socket.IO Javascript client API specifies both socket.connected and socket.disconnected. [The docs](https://socket.io/docs/v4/client-api/#socketconnected).

Why? Why not just one or the other? It's confusing. Is there actually some scenario where they are both true or both false (and what does that scenario mean)?

**Comment by darrachequesne** [comment, marked as answer]
That's a good question, actually... It was added a while back, in https://github.com/socketio/socket.io-client/commit/558f8b26266a868b98691bc7258b33c445f8c7e0

>  Is there actually some scenario where they are both true or both false?

No, I don't think this is possible.

**Reply by ryanwc** [reply]
Ah, ok, thanks. I am new to Socket.IO as of yesterday and was afraid I was missing something basic; guess not!

At least for me, having both of those is confusing. But maybe other people feel differently and/or some code now depend on both.

**Reply by darrachequesne** [reply]
> At least for me, having both of those is confusing

Yes, two properties that convey the same information is indeed confusing. I guess we can remove `disconnected` in the next major version.
